pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -478,10 +478,23 @@ func (ca *changeAggregator) Next() (rowenc.EncDatumRow, *execinfrapb.ProducerMet
 		if err := ca.tick(); err != nil {
 			var e kvevent.ErrBufferClosed
 			if errors.As(err, &e) {
-				// ErrBufferClosed is a signal that
-				// our kvfeed has exited expectedly.
+				// ErrBufferClosed is a signal that our kvfeed has exited expectedly. If
+				// so we need to make sure all events are cleared to the sink before
+				// exiting.
```

Comment:
I'm not sure this is the right place for a fix...
consider https://github.com/cockroachdb/cockroach/blob/master/pkg/ccl/changefeedccl/kvfeed/kv_feed.go#L145-L145
We drain kv_feed buffers before we exit.  Perhaps we should do the same here.
Another possibility is to emit EXIT boundary type so that changefeed naturally exits.

Reply:
I understand, but what I don't understand is why can't we drain the buffer at kvfeed -- is that not sufficient -- just like we do in case of schema changes?